packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java

```diff
@@ -30,14 +30,18 @@
    *
    * @param cameraProperties Collection of characteristics for the current camera device.
    * @param resolutionPreset Platform agnostic enum containing resolution information.
-   * @param cameraId Camera identifier of the camera for which to configure the resolution.
+   * @param cameraName Camera identifier of the camera for which to configure the resolution.
    */
   public ResolutionFeature(
-      CameraProperties cameraProperties, ResolutionPreset resolutionPreset, int cameraId) {
+      CameraProperties cameraProperties, ResolutionPreset resolutionPreset, String cameraName) {
```

Comment:
Why the change from int to string here, if it has to be an int in practice?

Reply:
Because Android supplies a list of camera identifies of type `String` and the whole API will work with the `String` type except for the `CamcorderProfile.hasProfile` and `CamcorderProfile.get` which demand the camera identifier to be an `int` (see Android doc [here](https://developer.android.com/reference/android/media/CamcorderProfile)). 

Since the `ResolutionFeature` is the only feature making use of the `CamcorderProfile` SDK we decided to leave the conversion in this class and use the `checkIsSupported` method to indicate if the feature is supported for the selected camera. For some external camera devices the camera identifier will be a `String` and for those the `ResolutionFeature` will not be supported.

Reply:
Makes sense, thanks!